libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -113,26 +115,48 @@ def batch(
         """
         configs = get_config_list(config, len(inputs))
 
+        def invoke(input: Input, config: RunnableConfig) -> Union[Output, Exception]:
```

Comment:
can't imagine that has any real performance implications but just in case, would it make sense to conditionally define invoke instead of checking within invoke (each time it's called). ie
```python
if return_exceptions:
  def invoke()...
else:
  def invoke()...
```